### PR TITLE
CollectionItem `href` property support

### PR DESCRIPTION
**Pull request by linuxenko**
Idk how it could be handled for all items (item identification based on `href` prop). Anyways it work just fine for me, maybe this `pull` can be helpful to identify the problem.

### CODE REVIEW

**Comment by isaiah** [comment]
Could you please squash the commits?

**Comment by linuxenko** [comment]
Didnot made "git squashing" before . Is it right ?

**Comment by isaiah** [comment]
No :smile: 

Check this out (https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/)

**Comment by linuxenko** [comment]
Strange ,[it](https://github.com/react-materialize/react-materialize/pull/59/commits/a023ce33e4bf5bce85b71e7863741178e0d77a58) were created by following these tutorial.

**Comment by isaiah** [comment]
`eslint` is not happy :(

**Comment by isaiah** [comment]
@linuxenko do you need a hand on squashing?

**Comment by linuxenko** [comment]
@isaiah aha, i need help with it

File: examples/InputCheckbox.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ import Row from '../src/Row';
 
 export default
 <Row>
-    <Input name='group1' type='checkbox' value='red' label='Red' />
-    <Input name='group1' type='checkbox' value='yellow' label='Yellow' checked='checked' />
-    <Input name='group1' type='checkbox' value='green' label='Green' className='filled-in' checked='checked' />
+    <Input name='group1' type='checkbox' value='red' label='Red'  />
```
Comment (alextrastero) [inline examples/InputCheckbox.js]: no need for extra space

File: examples/InputCheckbox.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ import Row from '../src/Row';
...
+    <Input name='group1' type='checkbox' value='yellow' label='Yellow' defaultValue="checked" />
```
Comment (alextrastero) [inline examples/InputCheckbox.js]: please use singlequotes

File: examples/InputCheckbox.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ import Row from '../src/Row';
...
+    <Input name='group1' type='checkbox' value='green' label='Green' className='filled-in' defaultValue="checked" />
```
Comment (alextrastero) [inline examples/InputCheckbox.js]: singlequotes :)

File: src/Button.js [hunk 4]
```diff
@@ -66,7 +66,9 @@ class Button extends React.Component {
   renderIcon() {
     if (this.props.icon) {
       return <Icon>{this.props.icon}</Icon>;
-    } else {
+    } else if (this.props.faicon) {
+      return <i className={this.props.faicon}></i>
+    }else {
```
Comment (alextrastero) [inline src/Button.js]: space before else please

File: src/CollectionItem.js [hunk 5]
```diff
@@ -11,15 +11,24 @@ class CollectionItem extends React.Component {
...
-    let C = this.isAnchor() ? 'a' : 'li';
+
+    if (this.isAnchor()) {
+      return (
+          <a className={cx(classes)} href={this.props.href}>
+           {this.props.children}
+          </a>
+      )
+    }
+
```
Comment (alextrastero) [inline src/CollectionItem.js]: 2 spaces instead of 4 on this file please :)

File: src/Input.js [hunk 6]
```diff
@@ -125,6 +126,17 @@ class Input extends React.Component {
       if (React.Children.count(children) == 1) {
         icon = React.Children.only(children);
       }
+
+      switch(inputType) {
+        case 'checkbox':
+          //props.checked = !!this.state.value
```
Comment (alextrastero) [inline src/Input.js]: comment here?

**Comment by alextrastero** [comment]
Nice! 👍

**Comment by alextrastero** [comment]
@linuxenko do you want to try to squash your commits or shall we do it?
 - http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by linuxenko** [comment]
@alextrastero ,  i'm afraid about any `rebase` commands )) Please you do

**Comment by alextrastero** [comment]
@linuxenko i'm afraid I can't do it because you're not working on a branch...
Could you do the following from your master branch:
```
git checkout -b collection_item_href_support
git push origin collection_item_href_support
```
And then create a new pull request with this branch?
I could do this for you but then it wouldn't show on your github that you contributed to this project 💯

**Comment by alextrastero** [comment]
Then I can fetch your new *branch* squash and push 👍

**Comment by alextrastero** [comment]
@linuxenko but I really recommend you learn Git if you want to be part of the open source world :-)
This is a good way: https://try.github.io/levels/1/challenges/1

**Comment by linuxenko** [comment]
It thought 'stars' is a most important part of opensource ))) Thanks, i'll learn about it

**Comment by alextrastero** [comment]
merged here: https://github.com/react-materialize/react-materialize/pull/65

**Pull request closed without merging**